internal/test/e2e/cluster.go

```diff
@@ -332,12 +333,12 @@ func newTestCluster(t *testing.T, opts testClusterOptions) *TestCluster {
 	busShutdownFns = append(busShutdownFns, bStopFn)
 
 	// Create worker.
-	w, s3Handler, wShutdownFn, err := node.NewWorker(workerCfg, s3.Opts{}, busClient, wk, logger)
+	w, s3Handler, webhooks, wShutdownFn, err := node.NewWorker(workerCfg, s3.Opts{}, busClient, wk, workerAddr, logger)
```

Comment:
Instead of returning webhooks I'd return a `registerCacheFn` which has to be called on a bus. Makes it a bit more obvious to the caller that there is something that needs to be done apart from calling `NewWorker` for the caching to work properly.

If the worker doesn't already do that, that function could also refresh/init the cache after subscribing to make sure we don't run into any races. Ideally the cache would not work if the webhooks aren't registered and always fetch things on demand with a warning that says "cache disabled cause webhooks weren't registered".